src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -181,6 +189,102 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
             return None
         return table
 
+    def create_version_in_package(
+        self,
+        package_role: str,
+        package_name: str,
+        stage_fqn: str,
+        version: str,
+        label: str | None = None,
+    ):
+        """
+        Creates a new version in an existing application package.
+        @param package_role: Switch to this role while executing create version.
```

Comment:
The package role should always be necessary since we are altering application package. No db/schema/wh is needed I don't think. The current implementation of creating new version or adding a patch does not specify a warehouse.

Reply:
Frank has a fair point though, being consistent with argument order across the APIs exposed by the facade makes things easier for developers overall. As for the role being required, I'm not sure we should enforce it here. The caller has full context. If they know the current role is fine, then so be it.

Reply:
hmm I can see both sides of the argument for and against this method enforcing the role. But to stay consistent, I updated the order and made role optional :)